ledger/acctonline.go

```diff
@@ -364,13 +374,26 @@ func (ao *onlineAccounts) prepareCommit(dcc *deferredCommitContext) error {
 	// being updated multiple times. When that happen, we can safely omit the intermediate updates.
 	dcc.compactOnlineAccountDeltas = makeCompactOnlineAccountDeltas(deltas, dcc.oldBase, ao.baseOnlineAccounts)
 
-	ao.accountsMu.RUnlock()
-
 	dcc.onlineAccountExpiredRowids = expirations
 	dcc.expirationOffset = expirationOffset
 
 	dcc.genesisProto = ao.ledger.GenesisProto()
 
+	start, err := ao.roundParamsOffset(dcc.oldBase)
+	if err != nil {
+		return err
+	}
+	end, err := ao.roundParamsOffset(dcc.newBase)
+	if err != nil {
+		return err
+	}
+	dcc.onlineRoundParams = ao.onlineRoundParamsData[start+1 : end+1]
+	maxOnlineLookback := basics.Round(ao.maxOnlineLookback())
+	dcc.maxLookbackRound = basics.Round(0)
+	if dcc.newBase > maxOnlineLookback-1 {
+		dcc.maxLookbackRound = dcc.newBase + 1 - maxOnlineLookback
```

Comment:
use `SubSaturarate` without the condition
```suggestion
		dcc.maxLookbackRound = (dcc.newBase + 1).SubSaturarate(maxOnlineLookback)
```